### ISSUE TITLE
Generate crosswalk.csv table automatically with separated mappings

### ISSUE THREAD

**Issue by moranegg** [codemeta/codemeta#204]
The crosswalk table file is changed every time a contribution is made without guarantee that the data, already in it, didn't change.
The proposition is to have a separate file for each mapping that will be aggregated into the crosswalk.csv file with a simple script.
 
This suggestion will result with some data duplication but could be much easier to use and will be more adequate for machine readability.
 
Before opening a PR for it, I would like to hear your thoughts, comments or additions to this suggestion, the goal is to make the crosswalk table more stable.

**Comment by cboettig** [comment]
Yeah, agree :100: about this, would be much more robust to have a separate file for each vocabulary in the crosswalk.

What format did you have in mind?  Probably best to just have a separate `.csv` file for each.  T(echnically might be 'better' to have a JSON-LD context for each, but that will be difficult for some contributors to edit and can probably be generated on-the-fly from the csv files, though it will leave us guessing at some of the datatypes)

**Comment by ProgVal** [comment]
>  Probably best to just have a separate .csv file for each.

As in, a .csv with two columns? Sounds good!

> a JSON-LD context for each

What do you mean?
Can contexts be used to translate from non-JSON-LD data?

**Comment by cboettig** [comment]
> What do you mean?
  Can contexts be used to translate from non-JSON-LD data?

Technically no, but really a context is just a bunch of name mappings.  Anything you can write in a CSV format you can also write in JSON, and then a context is just a mapping of of key names.  (Manu Sporny's [short video](https://codemeta.github.io/jsonld/) does a good job of showing this, I think).

**Comment by cboettig** [comment]
but yeah, one `csv` of two columns (codemeta terms and existing vocab terms) for each existing vocab sounds like the way to go here.

**Comment by ProgVal** [comment]
Got it, thanks!

Using a JSON-LD context here would have the extra advantage of working with nested structures; but as its expressivity would be very limited (eg. NPM's `author` field allows strings that have to be parsed with a regexp), so it may not be worth the extra complexity.

**Comment by mbjones** [comment]
While I agree with the advantages here you listed, the main disadvantage is losing our ability to simply view the crosswalk as an integrated table without the integration script that requires us to build to see the integration.  Are you imagining the script would be run via a CI system and check in the integrated crosswalk, or users would have to run it themselves?  I think the integrated file should be in the repository if possible for easy viewing.

**Comment by moranegg** [comment]
I agree both script and table should be in the repository. The idea was to stop accepting PRs on the table itself which can cause major changes in other mappings.

**Comment by ProgVal** [comment]
@mbjones I find the CSV table in the repository to be hard to use, because GitHub limits its width (even on large screen).

Instead, it could be build from the mappings and displayed on the website (which is manually built by @cboettig as far as I understand), which would allow for easier viewing, eg. by always keeping the top row and left column on screen no matter how much you scroll down/right.
